src/expression_evaluator/lambda_evaluator.cpp

```diff
@@ -48,9 +48,29 @@ void ListLambdaEvaluator::init(const ResultSet& resultSet, ClientContext* client
     bindData = ListLambdaBindData{lambdaParamEvaluators, paramIndices, lambdaRootEvaluator.get()};
 }
 
+// TODO(Ziyi): we currently do not support evaluating a list with size > DEFAULT_VECTOR_CAPACITY in
+// lambda.
+static void validateListLen(std::vector<std::shared_ptr<ValueVector>> params) {
+    for (auto& param : params) {
+        if (param->dataType.getLogicalTypeID() == LogicalTypeID::LIST) {
```

Comment:
I think we have the guarantee the first parameter is always of LIST type so you can assert here.